### DISCUSSION TITLE
ProgressRing can't be used in Skia.Gtk

### DISCUSSION THREAD

**Discussion by zipgenius** [unoplatform/uno#9693] [Q&A]
### Current behavior

App is really simple because it is a test. This is the Shared MainPage.xaml code:

```xaml
<Page
    x:Class="App1.MainPage"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:local="using:App1"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    mc:Ignorable="d"    
    Background="{ThemeResource ApplicationPageBackgroundThemeBrush}">

    <Grid>
        <ProgressRing Maximum="100" Value="50" Width="80" Height="80"/>
    </Grid>
</Page>
```

It initially runs but ProgressRing doesn't appear; console warns "Microsoft.UI.Xaml.Controls.ProgressRing[0]
      ProgressRing control needs the Uno.UI.Lottie package to run properly."

After adding the following packages:

- Uno.WinUI.Lottie (it's a WinUI-based app, otherwise I would meet a conflict while compiling)
- SkiaSharp.Views.Uno.WinUI
- SkiaSharp.Skottie

The app doesn't compile with the following error (in italian):
```
Errore MSB3025	il file di origine "C:\Users\matte\.nuget\packages\skiasharp.nativeassets.webassembly\2.88.1\build\netstandard1.0\libSkiaSharp.a" è in realtà una directory. L'attività "Copia" non supporta la copia di directory.	App1.Skia.Gtk	C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Current\Bin\amd64\Microsoft.Common.CurrentVersion.targets	5128	"
```

Which means that libSkiaSharp.a is a directory and action "Copy" doesn't support copying a directory.

I'm not able to execute my application anymore.

### Expected behavior

The application should compile fine and the ProgressRing should appear as expected

### Anything else we need to know?

I even tried to clear the Nuget cache and to install again everything from scratch but without success.

**Comment by FrozDark** [comment]
Did you try the latest dev builds of uno?
Also try SkiaSharp latest preview (not stable)

Install them and try again
SkiaSharp 2.88.1-preview.108
SkiaSharp.Skottie 2.88.1-preview.108
SkiaSharp.Views.Uno.WinUI 2.88.1-preview.108

Me running this with latest dev builds
![125125161](https://user-images.githubusercontent.com/6670150/187624120-bd3d3b3c-4276-4b87-ae56-5211ea0fb472.jpg)

I'm faced an issue with the 2.88.1 stable buids of skiasharp (Don't know if it was the same as yours). But I didn't look into it and skipped that issue installing the preview ones

**Reply by jeromelaban** [reply]
@zipgenius it looks like packages are being used that are unrelated to Skia.Gtk. Could your [provide a binlog](https://platform.uno/docs/articles/uno-builds-troubleshooting.html) here ?

**Reply by jeromelaban** [reply]
Also, it seems there's an issue with SkiaSharp 2.88.1, where some of the Skottie internals are not correctly setup. Best is to use the version suggested by Uno (2.88.1-preview.79)

**Reply by zipgenius** [reply]
I reverted all SkiaSharp-related packages to version 2.88.1-preview.108 and everything is working fine now. It is clearly a SkiaSharp issue in 2.88.1

**Reply by jeromelaban** [reply]
Looks like some unused parts of the msbuild targets got included, this should fix it: https://github.com/mono/SkiaSharp/pull/2231

**Comment by zipgenius** [comment]
> 

I'll try for sure
